lua/lvim/core/cmp.lua

```diff
@@ -156,62 +156,66 @@ M.config = function()
       fields = { "kind", "abbr", "menu" },
       max_width = 0,
       kind_icons = lvim.icons.kind,
-      source_names = {
-        nvim_lsp = "(LSP)",
-        emoji = "(Emoji)",
-        path = "(Path)",
-        calc = "(Calc)",
-        cmp_tabnine = "(Tabnine)",
-        vsnip = "(Snippet)",
-        luasnip = "(Snippet)",
-        buffer = "(Buffer)",
-        tmux = "(TMUX)",
-        copilot = "(Copilot)",
-        treesitter = "(TreeSitter)",
-      },
-      duplicates = {
-        buffer = 1,
-        path = 1,
-        nvim_lsp = 0,
-        luasnip = 1,
+      source_settings = {
```

Comment:
@kylo252 I actually really like because there are already a few source settings sprinkled into the sources list, see copilot for an example:

```lua
sources = {
      {
        name = "copilot",
        -- keyword_length = 0,
        max_item_count = 3,
        trigger_characters =
-- etc.
```
So consolidating the rest of the settings in there make sense. However, as you can see, the sources object is a list so we can't index in as you suggested `cmp.sources[foo].options`. We could change it to that format i.e.

```lua
sources = {
  copilot = { 
    -- options for copilot
  }
  other_source = {
   -- other options
  }
```
This would give us a single place to add sources and configure them in a way that I thinks makes sense, of course we would need to update everything that access the sources list to handle the new format.